packages/proxy/lib/http/response-middleware.ts

```diff
@@ -512,6 +512,13 @@ const CopyResponseStatusCode: ResponseMiddleware = function () {
   this.next()
 }
 
+// Set custom status message/reason phrase from http response
+// https://github.com/cypress-io/cypress/issues/16973
+const SetResponseStatusMessage: ResponseMiddleware = function () {
+  this.res.statusMessage = this.incomingRes.headers['reason-phrase']?.toString() || this.res.statusMessage
```

Comment:
@kshastri I don't think that we care about the `reason-phrase` header - I couldn't find that documented anywhere, is it used by some software?

Also, can you combine this middleware with "CopyResponseStatusCode"? They do really similar things, no sense in having 2 middlewares to copy 1 line from the HTTP preamble.

Also, I believe you want `this.incomingRes.statusMessage`, or else you're copying from the outgoing res:

```suggestion
  this.res.statusMessage = this.incomingRes.statusMessage
```

Reply:
@flotwig - This has been resolved 
`this.res.statusMessage = this.incomingRes.statusMessage || ''`